ccx/api/jobs.go

```diff
@@ -0,0 +1,96 @@
+package api
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+)
+
+type jobType string
+
+const (
+	deployStoreJob    jobType = "JOB_TYPE_DEPLOY_DATASTORE"
+	modifyDbConfigJob jobType = "JOB_TYPE_MODIFYDBCONFIG"
+	destroyStoreJob   jobType = "JOB_TYPE_DESTROY_DATASTORE"
+)
+
+type jobStatus string
+
+const (
+	jobStatusUnknown  jobStatus = "JOB_STATUS_UNKNOWN"
+	jobStatusRunning  jobStatus = "JOB_STATUS_RUNNING"
+	jobStatusFinished jobStatus = "JOB_STATUS_FINISHED"
+	jobStatusErrored  jobStatus = "JOB_STATUS_ERRORED"
+)
+
+type jobs struct {
+	baseURL string
+	auth    authorizer
+}
+
+type jobsResponse struct {
+	Jobs  []jobsResponseJobItem `json:"jobs"`
+	Total int
+}
+
+type jobsResponseJobItem struct {
+	JobID  string    `json:"job_id"`
+	Type   jobType   `json:"type"`
+	Status jobStatus `json:"status"`
+}
+
+func (svc jobs) Await(ctx context.Context, storeID string, job jobType, wait time.Duration) (jobStatus, error) {
+	attempts := (wait / time.Second) * 2
```

Comment:
Please don't calculate the attempts - keep trying until the time has actually passed, including time waiting for requests, and time waiting between requests.